Session8/ZooManagement/Zoo.java

```diff
@@ -0,0 +1,122 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+
+/**zoo management which add cage in the zones and animals in the cages .
+ * 
+ * @author Aman Gautam
+ * 
+ * Dated 24/07/2019
+ *
+ */
+public class Zoo {
+	final int MAXNOZONES = 3; 
+	final int MAXNOCAGES = 3; 
+	final int MAXNOANIMALS = 3;
+	
+	Scanner sc = new Scanner(System.in);
+	
+	String zoneCategory;
+	int noOfCages;
+	boolean park,canteen ;
+	float weight =(float)22.2;
+	int countZone=0, age=22;
+	
+	//list of ZONES
+	private List<Zone> listOfZones = new ArrayList<>();
+		
+	private Cage cage ;
+	private Zone zone ;
+	private Animal animal;
+	/*
+	 * ZOne is add by user input and cages or also created based on user 
+	 * requirements.
+	 */
+	public void addZone() {
+		System.out.println("enter the category of zone : ");
+		zoneCategory = sc.next();
+		System.out.println("enter the number of cages in zone : ");
+		noOfCages = sc.nextInt();
+		System.out.println("want to add park in zone [yes/no]: ");
+		park = sc.nextBoolean();
+		System.out.println("want to add canteen in zone [yes/no]: ");
+		canteen = sc.nextBoolean();
+		listOfZones.add(new Zone(zoneCategory, noOfCages, park, canteen));
+		System.out.println("zone added successfully.. ");
+		listOfZones.add(zone =new Zone(zoneCategory,noOfCages, park, canteen));
+		zone.noOfCages=3;
+	}
+	
+	/*
+	 * Factory Design Pattern , used to create object of animal
+	 * type , it get animal type and animal name from user and
+	 * return objects of the same animal type 
+	 */
+	 public Animal getAnimalFactory(String animalType ,String name){  
+
+		if (animalType == null) {
+			throw new AssertionError("Animal type cannot be empty ");
+
+		} else if (animalType.equalsIgnoreCase("PEACOCK")) {
+			return new Peacock(name, weight, age);
+
```

Comment:
Avoid using multiple return statement.
Can be done like these:
create a private variable `Animal animalType`
Then in each if conditions just define this object : `animalType = new Peacock(name, weight, age)`
Then at the end: just return the animalType object.